lib/bots/versionbot.ts

```diff
@@ -845,15 +892,56 @@ export class VersionBot extends GithubBot.GithubBot {
                                 };
                                 this.flowdock.postToInbox(flowdockMessage);
                             }
-                            this.log(ProcBot.LogLevel.DEBUG, `MergePR: Merged ${owner}/${repo}#${prInfo.number}`);
+                            this.log(ProcBot.LogLevel.INFO, `MergePR: Merged ${owner}/${repo}#${prInfo.number}`);
                         });
                     }
                 });
             }
         });
     }
 
-    private reportError(error: VersionBotError) {
+    private validMaintainer(config: VersionBotConfiguration, event: GithubBotApiTypes.PullRequestEvent): void {
+        // If we have a list of valid maintainers, then we need to ensure that if the `ready-to-merge` label
+        // was added, that it was by one of these maintainers.
+        const maintainers = ((((config || {}).procbot || {}).githubbot || {}).versionbot || {}).maintainers;
+        if (maintainers) {
+            // Get the user who added the label.
+            if (!_.includes(maintainers, event.sender.login)) {
+                let errorMessage = `The \`${this.mergeLabel}\` label was not added by an authorised ` +
+                    'maintainer. Authorised maintainers are:\n';
+                _.each(maintainers, (maintainer) => errorMessage = errorMessage.concat(`* ${maintainer}\n`));
+                throw new Error(errorMessage);
+            }
+        }
+    }
+
+    private getConfiguration(owner: string, repo: string) {
+        // We need to get the config file, should it exist.
+        return this.retrieveGithubConfiguration(owner, repo).then((configuration: VersionBotConfiguration) => {
+            const config = configuration;
+
+            // If there is a config file and a minimum version, ensure this can run.
+            if (config) {
+                // If there's no minimum version, then we're fine.
+                const minimumVersion = (config.procbot || {}).minimum_version || true;
+                if (!minimumVersion) {
+                    throw new Error('${process.env.VERSIONBOT_NAME} cannot run as its version is less than the ' +
+                        'required minimum for the ${owner}/${repo} repository. ' +
+                        `This: ${process.env.npm_version_number}, Req: ${minimumVersion}`);
+                }
+            }
```

Comment:
Why doesn't the initial processConfiguration do this? It lives at the top-level in the config, and it feels like it should be checked at the top-level in the code, rather than per-bot.